Best practices/Debugging Source Generator.md

```diff
@@ -0,0 +1,95 @@
+## What are Source Generators?
```

Comment:
Should we rename this document to `Source Generators.md` to make it more general ?
 
Debugging can be just one part of this.

Reply:
Agreed, but in that case I would add a bit more general info about the source generators (e.g. how to start a new project, are there any limitations/requirements, how to publish them...). That would be beneficial for us since from what I've heard the official documentation on Source Generators is not that great.